### PR TITLE
COMMERCE-11251 Technical Task - Order Splitting Object Rule

### PR DESCRIPTION
**Pull request by ghost**
https://issues.liferay.com/browse/COMMERCE-11251

### CODE REVIEW

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-api/src/main/java/com/liferay/commerce/constants/CommerceConstants.java [hunk 1]
```diff
@@ -21,6 +21,8 @@
  */
 public class CommerceConstants {
 
+	public static final String KEY_COMMERCE = "commerce";
```
[the remark this answers is not in the document]
Comment (CrescenzoRegaLF) [reply modules/apps/commerce/commerce-api/src/main/java/com/liferay/commerce/constants/CommerceConstants.java:24]: it is a temporary key

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

File: modules/apps/commerce/commerce-api/src/main/java/com/liferay/commerce/constants/CommerceObjectActionExecutorConstants.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+	public static final String KEY_SPLIT_COMMERCE_ORDER_BY_CATALOG =
+		"split-commerce-order-by-catalog";
+
+	public static final String
+		SPLIT_COMMERCE_ORDER_BY_CATALOG_CONDITION_EXPRESSION =
+			"orderStatus = 10";
+
+	public static final String
+		SPLIT_COMMERCE_ORDER_BY_CATALOG_OBJECT_ACTION_DESCRIPTION =
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-api/src/main/java/com/liferay/commerce/constants/CommerceObjectActionExecutorConstants.java:33]: Why having a constant? It's a label that is localized in the object framework, I would just go with a string in the class defining the action

File: modules/apps/commerce/commerce-api/src/main/java/com/liferay/commerce/constants/CommerceObjectActionExecutorConstants.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+			"This action splits an order into supplier orders by catalog";
+
+	public static final String
+		SPLIT_COMMERCE_ORDER_BY_CATALOG_OBJECT_ACTION_LABEL =
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-api/src/main/java/com/liferay/commerce/constants/CommerceObjectActionExecutorConstants.java:37]: Same here

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:reevaluate:1377401_11192

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:reevaluate:1307401_6463

**Comment by CrescenzoRegaLF** [comment]
ci:reevaluate:1387401_7561

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:reevaluate:1307401_6482

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by pietrobocale** [comment]
Commerce tests seem to be fine

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:stop

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce

**Pull request closed without merging**